### PR TITLE
Add closing lazy thread safe producer

### PR DESCRIPTION
**Pull request by nuest**
and some minor other changes.

### CODE REVIEW

File: src/main/java/org/n52/iceland/util/LazyClosingThreadSafeProducer.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    public void destroy() {
+        if (nullType != null) {
+            close(nullType);
+        }
+    }
+
+    protected abstract T create()
+            throws ConfigurationError;
+
+    protected abstract void close(T nullType);
```
Comment (autermann) [inline src/main/java/org/n52/iceland/util/LazyClosingThreadSafeProducer.java]: `nullType`? Isn't the type the same for all `Locale`s?

File: src/main/java/org/n52/iceland/util/LocalizedLazyThreadSafeProducer.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+
+public abstract class LocalizedLazyThreadSafeProducer<T> extends LazyThreadSafeProducer<T> implements LocalizedProducer<T> {
+
+    private final LoadingCache<Locale, T> cache = CacheBuilder.newBuilder()
+            .build(new CacheLoader<Locale, T>() {
+                @Override
+                public T load(Locale key) {
+                    return create(key);
```
Comment (autermann) [inline src/main/java/org/n52/iceland/util/LocalizedLazyThreadSafeProducer.java]: Couldn't this be a method reference? `create()` only throws a `RuntimeException`...

File: src/main/java/org/n52/iceland/util/LazyClosingThreadSafeProducer.java (hunk 1 above)
Comment (nuest) [inline src/main/java/org/n52/iceland/util/LazyClosingThreadSafeProducer.java]: This one does not support locales. I just didn't know what to name the variable :-). Maybe ``item``, or ``element`` are better? Or what about ``product`` ?

File: src/main/java/org/n52/iceland/util/LocalizedLazyThreadSafeProducer.java (hunk 2 above)
Comment (nuest) [inline src/main/java/org/n52/iceland/util/LocalizedLazyThreadSafeProducer.java]: Copied this code from you as far as I remember ;-). I don't see what you mean though, even after reading up on method references. Sorry!

Comment (autermann) [inline src/main/java/org/n52/iceland/util/LocalizedLazyThreadSafeProducer.java]: I know, but I wrote that code before Java 8...

I thought `CacheLoader` would be a simple interface, so you could write `.build(this::create)`, but it's a abstract class, so nevermind.

File: src/main/java/org/n52/iceland/util/LazyClosingThreadSafeProducer.java (hunk 1 above)
Comment (autermann) [inline src/main/java/org/n52/iceland/util/LazyClosingThreadSafeProducer.java]: I always write `t` if the type variable is called `T`, but ´product` is also fine.

**Comment by autermann** [comment]
Could you merge your branch with `develop`?

**Comment by nuest** [comment]
Done

**Comment by nuest** [comment]
I also removed the license mapping file - the mapping is included in the mappings in the "cdn".

**Pull request merged**